aibolit/metrics/lcom4/lcom4.py

```diff
@@ -19,62 +19,182 @@
+                            if len(inv_funcs) > 0:
+                                for inv_argument in inv_funcs:
+                                    for method in full_method_exhaust:
+                                        if inv_argument == method[0]:
+                                            G.add_node(method_exhaust[0] + str(hash(method_exhaust[1])))
+                                            G.add_node((method[0]) + str(hash(method[1])))
+                                            G.add_edge(method_exhaust[0] + str(hash(method_exhaust[1])),
+                                                       (method[0]) + str(hash(method[1])), type='reference')
 
-        G = nx.Graph()
 
-        for i in methods:
-            G.add_node(i)
-        for i in fields:
-            G.add_node(i)
-        for key, val in graph.items():
-            for x in val:
-                G.add_edge(key, x)
 
+            #nx.draw(G, pos=nx.spring_layout(G), with_labels=True)
+            #plt.show()
+            break
         return nx.number_connected_components(G)
+
+    @staticmethod
+    def filter_node_by_level(node: Node, javalang_class) -> Generator[Node, None, None]:
+        for filtered_path, filtered_node in node.filter(javalang_class):
+            if LCOM4.get_class_depth(filtered_path) == 1:
+                yield filtered_path, filtered_node
+
+    @staticmethod
+    def get_class_depth(path: tuple) -> int:
+        class_level = 0
+        for step in path:
+            if isinstance(step, (ClassDeclaration, InterfaceDeclaration, MethodDeclaration)):
+                class_level += 1
+        return class_level
+
+    @staticmethod
+    def find_method_declaration(method_name: str, method_parameters: List[Node], method_nodes: List[MethodDeclaration]):
+        for method_node in method_nodes:
+            if method_node.name == method_name:
+                yield method_node
+
+    def has_field(self, field_name: str, method_node: MethodDeclaration):
+        node_fields = self.filter_node_by_level(method_node, LocalVariableDeclaration)
+        field_names: List[str] = [local_node.declarators[0].name for path, local_node in node_fields]
+        if field_name in field_names:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def exhaust_method_name(method_node: Node) -> Tuple[Any, List[dict]]:
+        parameter_list = []
+        name = method_node.name
+        for parameter in method_node.parameters:
+            parameter_list.append((parameter.name, parameter.type.name))
+        return name, tuple(parameter_list)
+
+    @staticmethod
+    def exhaust_field_name(field_node: Node) -> Tuple[str, str]:
+        name = field_node.declarators[0].name
+        parameter_list = []
+        try:
+            parameter_list = ('type', field_node.type.name)
+        except AttributeError:
+            pass
```

Comment:
In case of error, the function returns None? then you have the wrong function signature